src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -0,0 +1,120 @@
+                                          const Robot& robot)
+{
+    robot_body_def.type = b2_dynamicBody;
+    robot_body_def.position.Set(robot.position().x(), robot.position().y());
+    robot_body_def.linearVelocity.Set(robot.position().x(), robot.position().y());
+    robot_body_def.angle           = robot.orientation().toRadians();
+    robot_body_def.angularVelocity = robot.angularVelocity().toRadians();
+
+    robot_body = world->CreateBody(&robot_body_def);
+}
+
+void PhysicsRobot::setupRobotPhysicsBody(const Robot& robot)
+{
+    // To create a polygon that approximates the shape of a robot, we find
+    // one of the points at the edge of the flat face at the front of the robot,
+    // and sweep around the back of the robot evenly distributing vertices until
+    // we reach the other edge of the flat face.
+
+    const unsigned int num_robot_body_vertices = b2_maxPolygonVertices;
+    b2Vec2 robot_body_vertices[num_robot_body_vertices];
+
+    Angle angle_to_edge_of_flat_face_relative_to_robot_orientation =
+        Vector(DIST_TO_FRONT_OF_ROBOT_METERS, FRONT_OF_ROBOT_WIDTH / 2.0).orientation();
+    Angle global_angle_to_edge_of_flat_face =
+        angle_to_edge_of_flat_face_relative_to_robot_orientation + robot.orientation();
+
+    Angle angle_to_sweep_across =
+        Angle::full() - (2 * angle_to_edge_of_flat_face_relative_to_robot_orientation);
+    Angle angle_increment = angle_to_sweep_across / (num_robot_body_vertices - 1);
+
+    for (unsigned int i = 0; i < num_robot_body_vertices; i++)
+    {
+        Angle angle_to_vertex = global_angle_to_edge_of_flat_face + (i * angle_increment);
+        // Shapes are defined relative to the body they are added to, so we do everything
+        // relative to the centre of the robot body, which is treated as (0, 0)
+        Vector vector_from_body_to_vertex =
+            Vector::createFromAngle(angle_to_vertex).normalize(ROBOT_MAX_RADIUS_METERS);
+        Point vertex           = Point(0, 0) + vector_from_body_to_vertex;
+        robot_body_vertices[i] = createVec2(vertex);
+    }
+
+    robot_body_shape.Set(robot_body_vertices, num_robot_body_vertices);
+    robot_fixture_def.shape = &robot_body_shape;
+
+    robot_fixture_def.density = 1.0;
+    // This is a somewhat arbitrary value. Collisions with robots are not perfectly
+    // elastic. However because this is an "ideal" simulation and we generally don't care
+    // about the exact behaviour of collisions, getting this value to perfectly match
+    // reality isn't too important.
+    robot_fixture_def.restitution = 0.5;
+    robot_fixture_def.friction    = 0.0;
+
+    robot_body->CreateFixture(&robot_fixture_def);
+}
+
+void PhysicsRobot::setupChickerPhysicsBody(const Robot& robot)
+{
+    // Create a very thin rectangle where the dribbler and chicker are on the
+    // flat face of the robot
+    robot_chicker_shape.SetAsBox(
+        0.001, DRIBBLER_WIDTH / 2.0,
```

Comment:
Nit:
Maybe create a constant for the thickness here

Reply:
I don't think thickness is important because this is just for simulation. We have no use for the thickness of the dribbler IRL, and it's harder to define.

What do you think?

Reply:
Sounds good to me. 

I was looking at it as more of a legibility thing